awx/main/tasks.py

```diff
@@ -867,6 +869,24 @@ def run(self, pk, isolated_host=None, **kwargs):
                 status, rc = manager_instance.run(instance, isolated_host,
                                                   kwargs['private_data_dir'],
                                                   kwargs.get('proot_temp_dir'))
+            elif hasattr(self, 'management_command'):
+                try:
+                    old_env = os.environ.copy()
+                    for k, v in env.items():
+                        os.environ[k] = v
+                    import_logger = logging.getLogger('awx.main.commands.inventory_import')
+                    old_stdout = import_logger.handlers[0].stream
+                    import_logger.handlers[0].stream = stdout_handle
+                    call_command(self.management_command, **command_args_to_kwargs(args))
```

Comment:
Now that we're removing pexpect usage, aren't we losing timeout support?  One of the benefits of our pexpect handler loop is that we can spawn a subprocess and gracefully handle timeouts.

Reply:
Maybe we could use subprocess or something? Or maybe that's too much work for too little gain. One thing I could do is to put this in the subprocess fork, and see if the timing gain still remains. I also think we would need to fully integrate with the main `run_pexpect` loop.

All very good points you raised here.